sklearn/impute/_knn.py

```diff
@@ -195,17 +205,14 @@ def fit(self, X, y=None):
         self : object
             The fitted `KNNImputer` class instance.
         """
+        self._validate_params()
         # Check data integrity and calling arguments
         if not is_scalar_nan(self.missing_values):
             force_all_finite = True
         else:
             force_all_finite = "allow-nan"
             if self.metric not in _NAN_METRICS and not callable(self.metric):
                 raise ValueError("The selected metric does not support NaN values")
```

Comment:
With my above suggested change of validating `self.metric` w/ `_NAN_METRICS, we can get rid of this entirely along w/ any tests checking for this in `test_knn.py`
```suggestion
```